docs/source/index.rst

```diff
@@ -67,8 +65,8 @@ parallelize one function at a time and not having to convert their whole workflo
 The same code can also be executed inside a jupyter notebook directly which enables an interactive development process.
 
 The standard `concurrent.futures.Executor <https://docs.python.org/3/library/concurrent.futures.html#module-concurrent.futures>`_
-interface is extended by adding the option :code:`cores_per_worker=2` to assign multiple MPI ranks to each function call.
-To create two workers :code:`max_workers=2` each with two cores each requires a total of four CPU cores to be available.
+interface is extended by adding the option :code:`max_workers=2` to assign two MPI ranks to each function call.
+To give each MPI rank two cores, we set :code:`cores_per_worker=2`, which requires a total of four CPU cores to be available.
```

Comment:
I was guessing here that the meanings of `cores_per_worker` and `max_workers` was swapped, but maybe I was wrong. Just wanted to draw attention to it.

Reply:
Oh I had forgotten that `max_workers` was part of the `concurrent.futures` interface. But it looks like actually that parameter is only defined by the concrete subclasses, and not `concurrent.futures.Executor` itself? In which case couldn't you have a completely different parameter with a different name and meaning? To be honest I'm still not certain what `max_workers` and `cores_per_worker` do, although I think if I tried it out it would make sense. In the example it might be nice to have different numbers for `max_workers` and `cores_per_worker` to help distinguish.

Reply:
I reverted my changes to this section.

Reply:
I am a bit confused, as both `concurrent.futures.ThreadPoolExecutor(max_workers=5)` and  `concurrent.futures.ProcessPoolExecutor(max_workers=5)` implement the `max_workers` parameter. For me the `max_workers` gives the number of python functions which can be executed in parallel. This is extended in the `pympipool.Executor` by supporting functions which use MPI internally via the `mpi4py` interface. So to be able to execute 5 functions in parallel each with 2 cores, the `pympipool.Executor` can be specified with `max_workers=5` and `cores_per_worker=2` resulting in a total of 10 cores being used.